app/controllers/concerns/application_shared.rb

```diff
@@ -150,6 +150,6 @@ def virtual_column_scope(base_scope)
     select_method = "select_#{virt_column}"
     model = respond_to?(:model_of_controller) ? model_of_controller : resource_class
     return if virt_column.blank? || model.columns_hash[virt_column] || !base_scope.virtual_column_scopes.include?(select_method.to_sym)
-    base_scope.public_send(select_method).search_for(params[:search]).order(params[:order])
+    base_scope.public_send(select_method).search_for(params[:search]).reorder(params[:order])
```

Comment:
This already considers `unless params[:order].blank?` since if it's empty the method return earlier and the logic returns to the previous (current) state ignoring virtual column.